src/include/OSL/rendererservices.h

```diff
@@ -189,6 +206,47 @@ class OSLEXECPUBLIC RendererServices {
                          const EncodedType* arg_types, uint32_t arg_values_size,
                          uint8_t* arg_values);
 
+    /// Builds a free function to provide a value for a given attribute.
+    /// This occurs at shader compile time, not at execution time.
+    ///
+    /// @param group
+    ///     The shader group currently requesting the attribute.
+    ///
+    /// @param is_object_lookup
+    ///     True if an object name was specified, even if the value is not
+    ///     known at compile time.
+    ///
+    /// @param object_name
+    ///     The object name, or nullptr if the value is not specified or it
+    ///     is not known at compile time.
+    ///
+    /// @param attribute_name
+    ///     The attribute name, or nullptr if the value is not known at
+    ///     compile time.
+    ///
+    /// @param is_array_lookup
+    ///     True if the attribute lookup provides an index.
+    ///
+    /// @param array_index
+    ///     The array index, or nullptr if the value is not specified or it
+    ///     is not known at compile time.
+    ///
+    /// @param type
+    ///     The type of the value being requested.
+    ///
+    /// @param derivatives
+    ///     True if derivatives are also being requested.
+    ///
+    /// @param spec
+    ///     The built attribute getter. An empty function name is interpreted
+    ///     as a missing attribute.
+    ///
+    virtual void
+    build_attribute_getter(ShaderGroup& group, bool is_object_lookup,
```

Comment:
Last one: group? const or not? Do we expect it may be modified by a possible  reasonable implementation of build_attribute_getter?

Reply:
Good question, by this point in the code we are jitting so I'd expect all modifications to group to have already been completed. So in that case I'd vote const.

Most of the `ShadingSystem` functions to access `ShaderGroup` info already accept `const ShaderGroup`s. There look to be two exceptions: `ShadingSystem:getattribute(ShaderGroup*, string_view, TypeDesc)` and `ShadingSystem::find_symloc(ShaderGroup*, ustring)` which would seem reasonable to access within these functions which might also need to get the const treatment if possible.

Reply:
It seems like those could be const? But this feels like a rathole.

Let's make the group& parameter to build_attribute_getter be const as well. But don't change SS::getattribute or find_symloc in this PR. We'll look at that later, and change down the road if we need to.